DSharpPlus.UnifiedCommands/Application/Conditions/IApplicationCondition.cs

```diff
@@ -0,0 +1,8 @@
+using DSharpPlus.Entities;
+
+namespace DSharpPlus.UnifiedCommands.Application.Conditions;
+
+public interface IApplicationCondition
+{
+    public Task<bool> InvokeAsync(DiscordInteraction interaction, DiscordClient client);
```

Comment:
if a method isn't supposed to be invoked by user code, we should consider making it a ValueTask